lib/project_types/extension/features/argo_config.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module Extension
+  module Features
+    class ArgoConfig
+      CONFIG_FILE_NAME = 'extension.config.yml'
+
+      class << self
+        def parse_yaml(context, permitted_keys = [])
+          file_name = File.join(context.root, CONFIG_FILE_NAME)
+
+          return {} unless File.size?(file_name)
+
+          require 'yaml' # takes 20ms, so deferred as late as possible.
+          begin
+            config = YAML.load_file(file_name)
+
+            # `YAML.load_file` returns nil if the file is not empty
+            # but does not contain any parsable yml data, e.g. only comments
+            # We consider this valid
+            return {} if config.nil?
+
+            unless config.is_a?(Hash)
+              raise ShopifyCli::Abort, ShopifyCli::Context.message('core.yaml.error.not_hash', CONFIG_FILE_NAME)
+            end
+
+            config.transform_keys!(&:to_sym)
+            assert_valid_config(config, permitted_keys) unless permitted_keys.empty?
+
+            config
+          rescue Psych::SyntaxError => e
+            raise(
+              ShopifyCli::Abort,
+              ShopifyCli::Context.message('core.yaml.error.invalid', CONFIG_FILE_NAME, e.message)
+            )
+          end
+        end
+
+        private
+
+        def assert_valid_config(config, permitted_keys)
+          unpermitted = config.keys.select do |k|
+            !permitted_keys.include?(k)
+          end
+
+          unless unpermitted.empty?
+            raise(
+              ShopifyCli::Abort,
+              ShopifyCli::Context.message(
+                'features.argo.config.unpermitted_keys',
+                CONFIG_FILE_NAME,
+                unpermitted.map { |k| "\n- #{k}" }.join
```

Comment:
to produce an output like 
![image](https://user-images.githubusercontent.com/474248/92675248-ebc54880-f2ec-11ea-86ce-d7b26250986f.png)

Maybe that can be done better?

Reply:
> Is this a screenshot of the actual output while tophatting?
Yep.

> I think the following might make sense

Just to be sure: The only difference in your suggestion compared to the screenshot is the backticks around the file name?

Reply:
As soon as we have public docs, we could also include a link to those. Which would probably help as well.

Reply:
That and removing 'config' at the end. This is more of a nitpick than anything else, just wanted to get UX to get some eyes on this.

Reply:
Missed that one! Glad I asked. 😅